packages/p2p/src/components/orders/order-details/order-details.jsx

```diff
@@ -88,37 +89,79 @@ OrderDetailsTimerBlock.propTypes = {
     order_details: PropTypes.object,
 };
 
-const OrderActionsBlock = ({ order_details }) => {
+const OrderActionsBlock = ({ cancelPopup, order_details, showPopup }) => {
     const {
         is_buyer,
         is_buyer_confirmed,
         is_pending,
+        offer_currency,
+        transaction_currency,
+        price_rate,
+        offer_amount,
     } = order_details;
     let buttons_to_render = null;
 
+    const cancelOrder = () => {
+        const options = {
+            title         : localize('Cancel this order?'),
+            message       : localize('There will be no refund after canceling the order. If you have paid, please do not cancel the order.'),
+            confirm_text  : localize('Cancel this order'),
+            onClickConfirm: cancelPopup,
+        }
+        showPopup(options);
+    }
+
+    const paidOrder = () => {
+        // TODO [p2p-order-api] call paid api
+        const options = {
+            title         : localize('Confirm this payment?'),
+            message       : localize('Make sure you have successfully sent the funds to the seller’s bank account or e-wallet mentioned above.'),
+            has_cancel    : true,
+            cancel_text   : localize('I didn\'t pay yet'),
+            confirm_text  : localize('I\'ve paid'),
+            onClickConfirm: cancelPopup,
+        }
+        showPopup(options);
+    }
+
+    const receivedFunds = () => {
+        const options = {
+            title            : localize('Have you received funds?'),
+            message          : localize('Make sure that you have logged in your bank account or other e-wallet to check the receipt.'),
+            need_confirmation: true,
+            offer: {
+                currency : offer_currency,
+                asset    : transaction_currency,
+                fix_price: price_rate,
+                amount   : offer_amount,
```

Comment:
should we standardize the naming for these?

Reply:
yes, we have to look into schema later on the other card to assemble all the flow

Reply:
can you add a todo comment instead then?

Reply:
done